test/unit/org/apache/cassandra/metrics/ThreadPoolMetricsTest.java

```diff
@@ -0,0 +1,311 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+import org.junit.Test;
+
+import org.apache.cassandra.Util;
+import org.apache.cassandra.concurrent.*;
+
+import static org.junit.Assert.*;
+
+public class ThreadPoolMetricsTest
+{
+    @Test
+    public void testJMXEnabledThreadPoolMetricsWithNoBlockedThread()
+    {
+        JMXEnabledThreadPoolExecutor executor = new JMXEnabledThreadPoolExecutor(2,
+                                                                                 Integer.MAX_VALUE,
+                                                                                 TimeUnit.SECONDS,
+                                                                                 new ArrayBlockingQueue<>(2),
+                                                                                 new NamedThreadFactory("ThreadPoolMetricsTest-1"),
+                                                                                 "internal");
+        testMetricsWithNoBlockedThreads(executor, executor.metrics);
+    }
+
+    @Test
+    public void testJMXEnabledThreadPoolMetricsWithBlockedThread()
+    {
+        JMXEnabledThreadPoolExecutor executor = new JMXEnabledThreadPoolExecutor(2,
+                                                                                 Integer.MAX_VALUE,
+                                                                                 TimeUnit.SECONDS,
+                                                                                 new ArrayBlockingQueue<>(2),
+                                                                                 new NamedThreadFactory("ThreadPoolMetricsTest-2"),
+                                                                                 "internal");
+        testMetricsWithBlockedThreads(executor, executor.metrics);
+    }
+
+    @Test
+    public void testSEPExecutorMetrics()
+    {
+        SEPExecutor executor = (SEPExecutor) new SharedExecutorPool("ThreadPoolMetricsTest-2").newExecutor(2,
+                                                                                                           "ThreadPoolMetricsTest-3",
+                                                                                                           "internal");
+
+        testMetricsWithNoBlockedThreads(executor, executor.metrics);
+    }
+
+    private static void testMetricsWithBlockedThreads(LocalAwareExecutorService threadPool, ThreadPoolMetrics metrics)
+    {
+        assertEquals(2, metrics.maxPoolSize.getValue().intValue());
+
+        spinAssertEquals(0, () -> metrics.activeTasks.getValue().intValue());
+        spinAssertEquals(0L, () -> metrics.completedTasks.getValue().longValue());
+        spinAssertEquals(0, () -> metrics.pendingTasks.getValue().intValue());
+        spinAssertEquals(0L, () -> metrics.currentBlocked.getCount());
+        spinAssertEquals(0L, () -> metrics.totalBlocked.getCount());
```

Comment:
Can we remove the unboxings and use method references instead of lambdas, here and in the other calls to `spinAssertEquals` below? Other than that, the test looks great to me.
```suggestion
        spinAssertEquals(0, metrics.activeTasks::getValue);
        spinAssertEquals(0L, metrics.completedTasks::getValue);
        spinAssertEquals(0, metrics.pendingTasks::getValue);
        spinAssertEquals(0L, metrics.currentBlocked::getCount);
        spinAssertEquals(0L, metrics.totalBlocked::getCount);
```